frontend/views/containers/proposals/RemoveMember.vue

```diff
@@ -1,64 +1,138 @@
 <template lang="pug">
 proposal-template(
+  ref='proposal'
   :title='L("Remove Member")'
-  :rule='{ value: 7, total: 10 }'
+  :rule='rule'
   :maxSteps='config.steps.length'
   :currentStep.sync='ephemeral.currentStep'
   @submit='submit'
 )
   .c-step(v-if='ephemeral.currentStep === 0' key='0')
     avatar.c-avatar(:src='member.picture')
-    i18n.is-title-4(tag='p' :args='{ name: member.displayName || member.name }') Remove {name} from your group
+    i18n.is-title-4(tag='p' :args='{ name: member.name }') Remove {name} from your group
+
+  banner-scoped(ref='formMsg' data-test='proposalError')
 </template>
 
 <script>
+import { mapState, mapGetters } from 'vuex'
+import sbp from '~/shared/sbp.js'
+import { CLOSE_MODAL } from '@utils/events.js'
+import L from '@view-utils/translations.js'
 import Avatar from '@components/Avatar.vue'
+import { PROPOSAL_REMOVE_MEMBER } from '@model/contracts/voting/proposals.js'
+import BannerScoped from '@components/banners/BannerScoped.vue'
 import ProposalTemplate from './ProposalTemplate.vue'
 
 export default {
   name: 'RemoveMember',
   components: {
     Avatar,
+    BannerScoped,
     ProposalTemplate
   },
+  props: {
+    username: String
+  },
   data () {
     return {
       ephemeral: {
         currentStep: 0
       },
       config: {
         steps: [
-          'AddMemberRule'
+          'RemoveMember'
         ]
       }
     }
   },
   created () {
-    console.log('TODO - Get username info')
+    if (!this.username) {
+      sbp('okTurtles.events/emit', CLOSE_MODAL)
+    }
   },
   computed: {
+    ...mapState([
+      'currentGroupId'
+    ]),
+    ...mapGetters([
+      'globalProfile',
+      'groupProfiles',
+      'groupSettings',
+      'groupShouldPropose',
+      'groupMembersCount'
+    ]),
     member () {
-      // TODO - connect to real store.
-      return {
-        id: 1000,
-        displayName: '[Attila the Hun]',
-        picture: '/assets/images/default-avatar.png'
-      }
+      return this.username ? this.globalProfile(this.username) : {}
```

Comment:
One of the issues with JS is that it lacks typing. To get around this, we use FlowType, but that also is insufficient. So the other trick we do is we use very descriptive name for everything (variables, computed properties, functions, etc.).

This computed property needs a better name, because `member` is too vague. It doesn't tell us anything about what is being returned. Let's therefore rename this to `memberGlobalProfile`. That then makes it clear what is being returned (the profile for member's global identity contract, as opposed to their group profile).

If their group profile should be returned instead, then it would be renamed to `memberGroupProfile` (and the `groupProfile` function would be used instead).

Reply:
member renamed to memberGlobalProfile 👍